docs/index.asciidoc

```diff
@@ -405,6 +410,69 @@ Cloud ID, from the Elastic Cloud web console. If set `hosts` should not be used.
 For more details, check out the
 {logstash-ref}/connecting-to-cloud.html[Logstash-to-Cloud documentation].
 
+[id="plugins-{type}s-{plugin}-data_stream"]
+===== `data_stream`
+
+* Value can be any of: `true`, `false` and `auto`
+* Default value depends on Logstash version.
+
+Defines whether data will be indexed into an Elasticsearch data stream.
+The other `data_stream_*` settings will only be used if this setting is enabled.
+Defaults to `false` in Logstash 7.x and `auto` starting in Logstash 8.0.
+
+The `auto` behavior means that whenever the supplied configuration is compatible
+with data streams Logstash will assume to be handling events as if `true` was set.
+
+[id="plugins-{type}s-{plugin}-data_stream_auto_routing"]
+===== `data_stream_auto_routing`
+
+* Value type is <<boolean,boolean>>
+* Default value is `true`.
+
+Automatically routes events by deriving the data stream name using specific event
+fields with the `%{data_stream.type}-%{data_stream.dataset}-%{data_stream.namespace}` format.
+
+If enabled, the `data_stream.*` event fields will take precedence over the
+`data_stream_type`, `data_stream_dataset`, and `data_stream_namespace` settings,
+but will fall back to them if any of the fields are missing from the event.
+
+[id="plugins-{type}s-{plugin}-data_stream_dataset"]
+===== `data_stream_dataset`
+
+* Value type is <<string,string>>
+* Default value is `generic`.
+
+The data stream dataset used to construct the data stream at index time.
+
```

Comment:
Are there other valid entries at this time?

Reply:
Believe anything is valid (as long as it's a valid name) in ES. 
Only the `data_stream_type` is contrained with enumerated values.